src/encoder.rs

```diff
@@ -321,7 +321,7 @@ impl SourceBlockEncoder {
         for i in 0..packets {
             let tuple = intermediate_tuple(start_encoding_symbol_id + i, lt_symbols, sys_index, p1);
             result.push(EncodingPacket::new(
-                PayloadId::new(self.source_block_id, start_encoding_symbol_id + i),
+                PayloadId::new(self.source_block_id, self.source_symbols.len() as u32 + i),
```

Comment:
I think this needs to include `+ start_repair_symbol_id` here

Reply:
You are right, thanks a lot for spotting that.

I've changed this and added a test that would have caught that bug.